### ISSUE TITLE
Encoder crashed when the SpeedControlFlag is set

### ISSUE THREAD

**Issue by Cidana-Developers** [OpenVisualCloud/SVT-AV1#539]
commit id: d0f64ff
build type: release
platform: ubuntu 16.04
Steps:
```
./SvtAv1EncApp -i ../../test/vectors/kirland_640_480_30.yuv -w 640 -h 480 -speed-ctrl 1 -o ../../../test/recon.yuv stat-report 1 -b ../../../test/output.ivf
-------------------------------------------
SVT-AV1 Encoder
SVT [version]:  SVT-AV1 Encoder Lib v0.6.0
SVT [build]  :  GCC 5.4.0        64 bit
LIB Build date: Aug 13 2019 17:28:34
-------------------------------------------
Number of logical cores available: 12
Number of PPCS 72
-------------------------------------------
SVT [config]: Main Profile      Tier (auto)     Level (auto)
SVT [config]: EncoderMode                                                       : 8
SVT [config]: EncoderBitDepth / EncoderColorFormat / CompressedTenBitFormat     : 8 / 1 / 0
SVT [config]: SourceWidth / SourceHeight                                        : 640 / 480
SVT [config]: FrameRate / Gop Size                                              : 30 / 32
SVT [config]: HierarchicalLevels / BaseLayerSwitchMode / PredStructure          : 4 / 0 / 2
SVT [config]: BRC Mode / QP  / LookaheadDistance / SceneChange                  : CQP / 50 / 33 / 0
-------------------------------------------
Encoding          Segmentation fault (core dumped)
```

**Comment by DesWurstes** [comment]
Even though I'm on macOS:

(Manual) Stack trace:

`eb_system_resource_ctor` calls `EB_NEW` many times, then `eb_system_resource_ctor`, `EbMuxingQueueCtor`, which calls `EbFifoCtor` multiplied by thread count, then `EB_CREATE_SEMAPHORE`, which creates ~250 semaphores before [this](https://github.com/OpenVisualCloud/SVT-AV1/blob/2f4852a031277cfb7cfd2f07c7a4f41c8591fcc7/Source/Lib/Common/Codec/EbThreads.c#L215) gives [code 24, too many files open by executable](https://stackoverflow.com/questions/24862733). (Note that in Linux it crashes earlier, before execute stack leaves `eb_create semaphore` because of [this line](https://github.com/OpenVisualCloud/SVT-AV1/blob/2f4852a031277cfb7cfd2f07c7a4f41c8591fcc7/Source/Lib/Common/Codec/EbThreads.c#L203), while on macOS it warns by printing the `errno` and `allocate memory failed, at Source/Lib/Common/Codec/EbSystemResourceManager.c, L29`) The problem is that there shouldn't be any malloc's without NULL check, which would at least prevent the segfault.

In `EbEncHandle.c`, `eb_system_resource_ctor` is called by `EB_NEW` many times, so is `EbFifoCtor`, but `EbFifoDctor` wasn't ever called, so the error is caused by un-reused or undestructed semaphores, reaching the limit of 255 (or 256) on my system.

LLDB output on my macOS (the same thing printed multiple times by different threads):
```
at EbSystemResourceManager.c:84:46
   81  	    fifoPtr->last_ptr = (fifoPtr->first_ptr == fifoPtr->last_ptr) ? (EbObjectWrapper*)EB_NULL : fifoPtr->last_ptr;
   82  	
   83  	    // Update head of BufferPool
-> 84  	    fifoPtr->first_ptr = fifoPtr->first_ptr->next_ptr;
   85  	
   86  	    return return_error;
   87  	}
```

What's also weird is #538 causes the same segfault on my computer.

**Comment by JackZhouVCD** [comment]
issue fixed by #544
